gammapy/maps/axes.py

```diff
@@ -2125,3 +2126,177 @@ def from_gti(cls, gti, name="time"):
+        idx = np.round(pix).astype(int)
+        return self._labels[idx]
+
+    def coord_to_idx(self, coord, **kwargs):
+        """Transform labels to indices
+
+        If the label is not present an error is raised.
+
+        Parameters
+        ----------
+        coord : `~astropy.time.Time`
+            Array of axis coordinate values.
+
+        Returns
+        -------
+        idx : `~numpy.ndarray`
+            Array of bin indices.
+        """
+        coord = np.array(coord)[..., np.newaxis]
+        is_equal = coord == self._labels
+
+        if not np.all(np.any(is_equal, axis=-1)):
+            label = coord[~np.any(is_equal, axis=-1)]
+            raise ValueError(f"Not a valid label: {label}")
+
+        return np.argmax(is_equal, axis=-1)
+
+    def coord_to_pix(self, coord):
+        """Transform from axis labels to pixel coordinates.
+
+        Parameters
+        ----------
+        coord : `~numpy.ndarray`
+            Array of axis label values.
+
+        Returns
+        -------
+        pix : `~numpy.ndarray`
+            Array of pixel coordinate values.
+        """
+        return self.coord_to_idx(coord).astype("float")
+
+    def pix_to_idx(self, pix, clip=False):
+        """Convert pix to idx
+
+        Parameters
+        ----------
+        pix : tuple of `~numpy.ndarray`
+            Pixel coordinates.
+        clip : bool
+            Choose whether to clip indices to the valid range of the
+            axis.  If false then indices for coordinates outside
+            the axi range will be set -1.
+
+        Returns
+        -------
+        idx : tuple `~numpy.ndarray`
+            Pixel indices.
+        """
+        if clip:
+            idx = np.clip(pix, 0, self.nbin - 1)
+        else:
+            condition = (pix < 0) | (pix >= self.nbin)
+            idx = np.where(condition, -1, pix)
+
+        return idx
+
+    @property
+    def center(self):
+        """Center of the label axis"""
+        return self._labels
+
+    @property
+    def edges(self):
+        """Edges of the label axis"""
+        raise ValueError("A LabelMapAxis does not define edges")
+
+    @property
+    def bin_width(self):
+        """Bin width is unity """
+        return np.ones(self.nbin)
+
+    def __repr__(self):
+        str_ = self.__class__.__name__ + "\n"
+        str_ += "-" * len(self.__class__.__name__) + "\n\n"
+        fmt = "\t{:<10s} : {:<10s}\n"
+        str_ += fmt.format("name", self.name)
+        str_ += fmt.format("nbins", str(self.nbin))
+        str_ += fmt.format("node type", self.node_type)
+        str_ += fmt.format(f"labels", "{0}".format(list(self._labels)))
+        return str_.expandtabs(tabsize=2)
+
+    # TODO: could create sub-labels here using dashes like "label-1-a", etc.
+    def upsample(self, *args, **kwargs):
+        """Upsample axis"""
+        raise NotImplementedError("Upsampling a LabelMapAxis is not supported")
+
+    # TODO: could merge labels here like "label-1-label2", etc.
```

Comment:
For merging labels, I would prefer to call it `.merge_nodes()` of something similar instead of `downsample` - Downsampling seems to indicate it will similarly on all bins, whereas it might be reasonable to merge `A` + `B`, and then `C`+`D`+`E` and just keep `F`.

But this is of course for later. For now, I `NotImplementedError` is reasonable. (Or do we even need to have/expose the `upsample` and `downsample`  options?)

Reply:
The advantage of exposing it, is we can raise an `NotImplementedError`, otherwise it would fail e.g. with and `AttributeError` if `Map.downsample()` is called and a `LabelMapAxis` is present. 

Note that `.downsample()` is indeed the same for all bins, but I fully agree a more general `.merge_labels()` or `.resample()` method would be useful.